### PR TITLE
Integrating depth anything 3: PR1

### PR DESCRIPTION
**Pull request by manas1245agrawal**
## 📝 Description

**Fixes/Relates to:**  https://github.com/kornia/kornia/issues/3488

## 🛠️ Changes Made
- [x] Adds a file named common.py

---

## 🧪 How Was This Tested?
- [x] **Manual Verification:** 
The code was manually checked by the author and each function was running succesfully.

---

## 🕵️ AI Usage Disclosure
*Check one of the following:*
- [] 🟢 **No AI used.**
- [x] 🟡 **AI-assisted:** I used AI for boilerplate/refactoring but have manually reviewed and tested every line.
- [ ] 🔴 **AI-generated:** (Note: These PRs may be subject to stricter scrutiny or immediate closure if the logic is not explained).

---

## 🚦 Checklist
- [x] I am assigned to the linked issue (required before PR submission)
- [x] The linked issue has been approved by a maintainer
- [x] I have performed a **self-review** of my code (no "ghost" variables or hallucinations).
- [x] My code follows the existing style guidelines of this project.
- [x] I have commented my code, particularly in hard-to-understand areas.

This is the first stage of integrating the Depth Anything 3 model. 
In this PR I have added some useful utils for da3 which is needed by both the model and its backbone (dinov2)
It brings in - 
Attention,LayerScale, Block DropPath and MLP

### CODE REVIEW

**Comment by manas1245agrawal** [comment]
> @manas1245agrawal could you try to fix the pre commit issues?

Good evening @Meggison , I have fixed the pre commit error and the test has passed successfully. Any further review is very much appreciated. Thankyou sir

File: kornia/models/depth_estimation_3/common.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+class Attention(nn.Module):
+    def __init__(
+        self,
+        dim: int,
+        num_heads: int = 8,
+        qkv_bias: bool = False,
+        proj_bias: bool = True,
+        attn_drop: float = 0.0,
+        proj_drop: float = 0.0,
+        norm_layer: nn.Module = nn.LayerNorm,
```
Comment (Meggison) [inline kornia/models/depth_estimation_3/common.py]: Fix type check issues

**Comment by Meggison** [comment]
@manas1245agrawal are there any active tests before or that you included that shows the model and its components work to make sure its adaptation is faithful?

**Review by Meggison — CHANGES_REQUESTED** [review]
Fix type check issues

**Comment by manas1245agrawal** [comment]
> @manas1245agrawal are there any active tests before or that you included that shows the model and its components work to make sure its adaptation is faithful?

Okay thankyou for the review. Can you please elaborate more on the tests you mentioned me.

**Comment by Meggison** [comment]
@manas1245agrawal need a test file that checks model components, encoder, attention mechanism, gradient checks and others relevant to the specific model implementation. Might be good to reference the original paper or implementation for what to test
